presto-native-execution/presto_cpp/main/functions/README.md

```diff
@@ -0,0 +1,23 @@
+# Prestissimo: Dynamically Loading Function Library
+
+This library adds the ability to load User Defined Functions (UDFs) without having to fork and build Prestissimo through the use of shared libraries. The Prestissimo worker is to access said code. The dynamic functions are to be loaded upon running an instance of the presto server. In the presto server instance, it will search for any .so or .dylib files and load them using this library.
+
+## Getting started
+
+The Process can be roughly broken down into the steps below:
+### 1. Create a cpp file for your dynamic library
+For dynamically loaded function registration, the format followed is mirrored of that of built-in function registration with some noted differences. For instance, the below example function uses the extern "C" keyword to protect against name mangling. Additionally, a registry() function call is also necessary here.
+
+### 2. Register functions dynamically by creating .dylib or .so shared libraries and dropping them in a plugin directory. 
+These shared libraries may be made using CMakeLists like the following:
+```
+add_library(name_of_dynamic_fn SHARED TestFunction.cpp)
+target_link_libraries(name_of_dynamic_fn PRIVATE xsimd fmt::fmt velox_expression)
+```
+
+### 3. In the prestissimo worker's config.properties file, set the plugin.dir property
+This is where you will be dropping your shared library.
+```
+plugin.dir="User\Test\Path\plugin"
+```
+### 4. Upon start of the Presto Server, this directory will be scanned and all shared libraries will be attempted to be dynamically loaded when the worker or the sidecar process starts.
```

Comment:
suggested wording here:

### 4. When the worker or the sidecar process starts, it will scan the plugin directory and attempt to dynamically load all shared libraries.